cloudinit/config/cc_growpart.py

```diff
@@ -265,13 +265,15 @@ def device_part_info(devpath):
 
     # FreeBSD doesn't know of sysfs so just get everything we need from
     # the device, like /dev/vtbd0p2.
-    if util.is_FreeBSD():
-        freebsd_part = "/dev/" + util.find_freebsd_part(devpath)
-        m = re.search("^(/dev/.+)p([0-9])$", freebsd_part)
+    fpart = "/dev/" + util.find_freebsd_part(devpath)
```

Comment:
I think we still need a guard under this section otherwise I think linux devices could match here and unexpected return early. Can we use `util.is_BSD()`?

Another option would be to move `if not os.path.exists(syspath):` up here and then do the BSD-related stuff after checking that the Linux-specific path doesn't exist.